inspect_byproducts.py

```diff
@@ -0,0 +1,168 @@
+
+
+def inspect_byproducts(link, type, operator, input_string):
+    """
+    <Purpose>
+    A function which performs the inspection as described above depending on
+    various arguments.
+
+    <Arguments>
+      link:
+        the path to the link file
+
+      std:
+        whether to check stdout or stderr field
+
+      operator:
+        is | is not | contains | contains not
+
+      input_string:
+        the string to be checked
+
+    <Exceptions>
+      Raises KeyError, in case the field corresponding to the key
+      in the dictionary is empty
+
+    <Returns>
+      Boolean
+    """
+
+    imported_link = link_import.read_from_file(link)
+
+    std_out_err = imported_link.byproducts[type]
+
+    if operator == 'is':
+      if std_out_err == input_string:
+        return True
+
+    elif operator == 'is-not':
+      if std_out_err != input_string:
+        return True
+
+    elif operator == 'contains':
+      if std_out_err.find(input_string) != -1:
+        return True
+
+    elif operator == 'contains-not':
+      if std_out_err.find(input_string) == -1:
+        return True
+    else:
+      raise('Invalid Operator ' + operator + '. Valid operators: is | is-not |'
+        ' contains | contains-not')
+
+    return False
+
+
+def parse_args():
+    """
+    <Purpose>
+      A function which parses the user supplied arguments.
+
+    <Arguments>
+      None
+
+    <Exceptions>
+      None
+
+    <Returns>
+      Parsed arguments (args object)
+    """
+    parser = argparse.ArgumentParser(
+        description="Inspects the byproducts of a step")
+
+    lpad = (len(parser.prog) + 1) * " "
+
+    parser.usage = ("\n"
+                    "%(prog)s --link <path to link metadata>\n{0}"
+                    "[--type (--stdout | --stderr)]\n{0}"
+                    "[--operator (--is | --is-not | --contains | --"
+                    "contains-not)]\n{0}"
+                    "string\n\n"
+                    .format(lpad))
+
+    in_toto_args = parser.add_argument_group("in-toto-inspection options")
+
+    in_toto_args.add_argument("-l", "--link", type=str, required=True,
+                              help="Link metadata file to use for inspection"
+                              " of the step")
+
+    in_toto_args.add_argument("-t", "--type", choices=['stdout', 'stderr'],
+                              type=str, required=True, help="Whether stdout or"
+                              " stderr is a byproduct")
+
+    in_toto_args.add_argument("-o", "--operator", choices=['is', 'is-not',
+                              'contains', 'contains-not'], type=str,
+                              required=True, help="whether "
+                              "stdout or stderr is, is not,"
+                              "contains, contains not, the input string")
+
+    in_toto_args.add_argument("string", type=str,
+                              help="The string to which the return value "
+                              "should be compared")
```

Comment:
I'd suggest: "The string to compare with the specified byproduct in the specified link file"